### PR TITLE
Remove typescript

### PR DESCRIPTION
**Pull request by tinkertrain**
What it says in the title...

We have concluded that TypeScript has become more of a hindrance than an actual helpful tool for this project, so this PR removes the use of TypeScript, some notes:

- Files that house the components are renamed from .tsx to .jsx, you can use .js all the same but for consistency and a little visual help (most code editors will give these files a special react icon), use this extension.
- Other files should use .js (even tests that contain jsx, look at how easy is to identify the components file in the screenshot below)
- Use `propTypes` for prop documentation
- Fixes and adds tests where needed
- Updates Jest, Babel and ESLint configurations
- Removes extraneous or no longer used dependencies
- Updates Node to 12.13
- Removes all instances of:
	- `/* istanbul ignore next */`
	- `/* istanbul ignore if */`
	- `/* istanbul ignore else */`
- Updates the coverage thresholds to:
```
  coverageThreshold: {
    global: {
      branches: 90,
      functions: 95,
      lines: 95,
      statements: 95,
    },
  },
```
- Updates the jest configuration to remove the individual files listed for ignoring, leaving only glob patterns and added `/* istanbul ignore file */` to the corresponding files.

### Current coverage with the changes
Screen Shot 2020-02-19 at 12 46 39 PM

### Ideal file structure:
Avatar utils js — hsds-react 2020-02-18 13-28-02

### CODE REVIEW

File: docs/quick-start/README.md [hunk 1]
```diff
@@ -12,20 +12,20 @@ Or, even faster, you can run:
...
-Blue has a couple of `peerDependencies`:
+HSDS has a couple of `peerDependencies`:
 
 - [`react`](https://www.npmjs.com/package/react)
 - [`react-dom`](https://www.npmjs.com/package/react-dom)
 
 Make sure you have these installed in your project.
 
-Blue currently runs on **React version 15**. Work is being done to add compatibility with 16+. 💪
+Blue currently runs on **React version 16**.
```
Comment (cen10) [inline docs/quick-start/README.md]: `HSDS` instead of `Blue`?

**Comment by cen10** [comment]
This looks awesome. Do you think it's worth adding a section to the HSDS README that explains the ideal file structure and using the `jsx` extension only for components?

**Comment by tinkertrain** [comment]
> This looks awesome. Do you think it's worth adding a section to the HSDS README that explains the ideal file structure and using the `jsx` extension only for components?

Absolutely! Documentation is the next thing I will be focusing on, from individual components (porting to mdx) to contribution guides.

PS Documentation will not be a requirement to release v3

File: docs/quick-start/README.md (hunk 1 above)
Comment (tinkertrain) [reply docs/quick-start/README.md]: Nice catch :), all the guides and dcos will be reviewed and updated soon, I'll leave these changes for now

**Pull request merged**